<?xml version="1.0"?>

<launch>

  <arg name="UAV_NAME" default="$(env UAV_NAME)" />
  <arg name="RUN_TYPE" default="$(env RUN_TYPE)" />
  <arg name="behavior_name" default="fire_challenge_state_machine" />

  <arg name="log_enabled" default="False" />
  <arg name="log_folder" default="~/.flexbe_logs" />

  <arg if="$(eval arg('RUN_TYPE') == 'simulation')" name="config_file" value="$(find mrs_flexbe_behaviors)/config/fire_simulation.yaml" />
  <arg unless="$(eval arg('RUN_TYPE') == 'simulation')" name="config_file" value="$(find mrs_flexbe_behaviors)/config/fire_reality.yaml" />

  <group ns="$(arg UAV_NAME)">
    <node name="flexbe_onboard_behavior" pkg="flexbe_onboard" type="start_behavior" output="screen" respawn="true">
      <param name="~log_enabled" value="$(arg log_enabled)"/>
      <param name="~log_folder" value="$(arg log_folder)"/>
    </node>

    <node name="flexbe_behavior_launcher" pkg="flexbe_widget" type="be_launcher" output="screen" args="-b '$(arg behavior_name)'">
      <param name="robot_name" value="$(arg UAV_NAME)"/>
      <rosparam file="$(arg config_file)" />
    </node>

  </group>
</launch>
